### PR TITLE
Feat(tests): Add comprehensive unit tests for analysis and data fetching

### PR DESCRIPTION
**Pull request by mohavro**
**## PR Description:
- Added comprehensive unit test suite for `FormulaicAnalyzer` in `tests/unit/test_formulaic_analysis.py`.
- Added comprehensive unit test suite for `RealDataFetcher` in `tests/unit/test_real_data_fetcher.py`.
- Refactored security tests in `tests/integration/test_pr_agent_config_validation.py` for improved clarity and maintainability.
- Applied minor code formatting across several files.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [X] Code refactoring
- [ ] Performance improvement
- [ ] Test addition/update
- [ ] Dependency update

## Related to

Closes #738

## Changes Made

- Replaced unused key variable with _ in dict iteration at line 322
- Removed dead code blocks containing unused variable definitions and empty function stubs

## Testing

- [X] Unit tests pass locally
- [X] Integration tests pass locally
- [ ] Manual testing completed
- [ ] New tests added for new functionality

### Test Commands

```bash
# Commands used to test
pytest
npm test
```

### Code Quality

- [X] My code follows the project's style guidelines
- [ ] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### Documentation

- [ ] I have updated the README.md (if needed)
- [ ] I have updated the CHANGELOG.md
- [ ] I have updated relevant documentation files
- [X] I have added/updated docstrings

### Dependencies

- [X] I have checked for dependency conflicts
- [ ] I have updated requirements.txt (if needed)
- [ ] I have updated package.json (if needed)

### Branch Management

- [X] My branch is up to date with main
- [X] I will delete this branch after merge
- [ ] This branch has a descriptive name
- [X] This PR has been open for less than 2 weeks

## Additional Notes

This PR will only address issues related to the PR description. 

---

**For Reviewers**: Please ensure the branch is deleted after merging this PR to maintain repository hygiene.

**Related Documentation**:
- [Contributing Guidelines](../CONTRIBUTING.md)
- [Branch Cleanup Guidelines](../BRANCH_CLEANUP_QUICK_REFERENCE.md)

---
## Summary by cubic
Removed unused iteration variables and simplified secret scanning; added unit tests for formulaic analysis and the real data fetcher. No production behavior changes.

- **Refactors**
  - Replaced unused dict loop variables with "_" and removed dead code stubs.
  - Consolidated secret markers and inline credential regex into small helpers in tests/integration/test_pr_agent_config_validation.py to reduce branching and duplication.
  - Fixed indentation/docstrings and applied formatting across modules without altering runtime logic.

<sup>Written for commit d9fbef52d882dfdbea70a9e22e39e4a132e0cde8. Summary will update on new commits. <a href="https://cubic.dev/pr/DashFin-FarDb/financial-asset-relationship-db/pull/744">Review in cubic</a></sup>

___

### **Auto-created Ticket**

[#745](https://github.com/DashFin-FarDb/financial-asset-relationship-db/issues/745)

### **Description**
- Refactored methods in `formulaic_analysis.py` to improve clarity and maintainability.
- Enhanced `create_real_database` in `real_data_fetcher.py` for better network handling.
- Added comprehensive unit tests for the `RealDataFetcher` class and its methods.
- Included edge case handl
...(truncated)

___

## **CodeAnt-AI Description**
Add comprehensive tests for formula analysis and real data fetching; tighten small CLI and formatting behaviors

### What Changed
- Added large suites of unit tests covering FormulaicAnalyzer behavior across equities, bonds, commodities, currencies, portfolio theory, correlation/risk formulas, cross-asset relationships, and edge/regression cases so analyze_graph and its helpers are exercised end-to-end.
- Introduced comprehensive unit tests for RealDataFetcher: initialization modes, cache save/load, network fallback behavior, serialization/deserialization of assets/events/graphs, and error cases for corrupted caches.
- Standardized many existing tests (workflow validator, database, repository) to use consistent string writing and single-line assertions to avoid incidental syntax/formatting flakes.
- Minor user-visible adjustments:
  - Server CLI now prints a single-line missing-dependency message on exit.
  - analyze_graph summary and formula outputs are produced consistently (summary fields present and r_squared averages bounded).

### Impact
`✅ More formula extraction coverage`
`✅ More real-data fetcher coverage`
`✅ Fewer workflow/test syntax failures`
<details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

**Comment by mohavro** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/dashfin.fardb/pr-checks/5f227288-1929-47cd-8e8d-06bffe9198c7?refs=1dd67a6612e85025b358b5b6f36b6b85aa2bb0d5&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: tests/integration/test_pr_agent_config_validation.py [hunk 1]
```diff
@@ -232,254 +249,171 @@ def _iter_string_values(obj):
...
+            for item in obj:
+                scan(item, suspected)
+
+        elif isinstance(obj, str):
+            stripped = obj.strip()
+            kind = _classify_stripped(stripped)
+            if kind:
+                suspected.append((kind, stripped))
+
+        def test_no_hardcoded_credentials(pr_agent_config):
```
[the remark this answers is not in the document]
Comment (mohavro) [reply tests/integration/test_pr_agent_config_validation.py]: @github-code-quality can you suggest code for fix?

**Comment by mohavro** [comment]
@qodo-merge-pro review pr
